shared/src/vyxal/ListHelpers.scala

```diff
@@ -854,6 +854,21 @@ object ListHelpers:
     current
   end reduce
 
+  def reshape(iterable: VList, shape: Seq[VNum]): VAny =
+    var index = -1
+    def nextElement(): VAny =
+      index += 1
+      if !iterable.isDefinedAt(index) then
+        index =
+          0 // avoid %= iterable.length to hopefully work with infinite lists
+      iterable(index)
+    def go(shape: Seq[Int]): VAny =
+      if shape.isEmpty then iterable(0)
+      else if shape.length == 1 then
+        VList.from((0 until shape.head).map { _ => nextElement() })
+      else VList.from((0 until shape(0)).map { _ => go(shape.tail) })
```

Comment:
`VList.fill` should work here (or at least, `VList.from(Seq.fill(shape.head)(...))`)